pkg/innerring/innerring.go

```diff
@@ -344,11 +347,11 @@ func New(ctx context.Context, log *zap.Logger, cfg *viper.Viper) (*Server, error
 		}
 	}
 
-	server.pubKey = crypto.MarshalPublicKey(&server.key.PublicKey)
+	server.pubKey = server.key.Bytes()
```

Comment:
I guess it should be 
```suggestion
	server.pubKey = server.key.PublicKey().Bytes()
```